### PR TITLE
Makes the crematorium able to be "broken" and fixed instead of destroyed.

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

The crematorium is no longer able to be destroyed but is instead able to be damaged into a broken state, at which point it must be repaired. This is a multi-stage process which requires first a new igniter, followed by the application of a welder until the paneling is fixed. The cremator button and cremator tray are themselves indestructible, as I can't currently think of an elegant solution to them and if I left them destructible this really wouldn't change anything.
It also technically adds a new icon (crema_broke), but this is identical to the cremator icon at the moment as I have no talent in spriting.
## Why It's Good For The Game

This was discussed a bit ago in Coding Chat as something that should be, especially with https://github.com/ParadiseSS13/Paradise/pull/19856 hopefully being added eventually. Ultimately, the Crematorium is a one-of-a-kind item on station which is a useful tool to destroy Changelings which is why the meta is frequently for changelings to destroy it, which would worsen if decapitation stopped being an effective tool. With this change, it is now able for a Changeling to disable the cremator for some time but for it to still be able to be repaired.
## Testing

image
image
image
## Changelog
:cl:
tweak: The Crematorium is no longer able to be destroyed, but is instead able to be damaged and repaired.
/:cl:

### CODE REVIEW

**Comment by McRamon** [comment]
While constructable crematorium is an issue, indestructible machine nad button is so much anti-sandbox. Perhaps there is a better way to go around it? Order it from cargo or something liek that

**Comment by Coolrune206** [comment]
Currently working on making the cremator able to be damaged, and afterwards be repaired.

File: code/game/objects/structures/morgue.dm [hunk 1]
```diff
@@ -354,12 +332,56 @@ GLOBAL_LIST_EMPTY(crematoriums)
...
+
+/obj/structure/crematorium/attackby(obj/item/P, mob/user, params)
 	if(is_pen(P))
 		rename_interactive(user, P)
 		add_fingerprint(user)
 		return
-
+	if(istype(P, /obj/item/assembly/igniter))
+		if(repairstate == 0)
+			user.visible_message("<span class='notice'>[user] replaces [src]'s igniter'.</span>", "<span class='notice'>You replace [src]'s damaged igniter. Now it just needs its paneling welded.</span>")
```
Comment (lewcc) [inline code/game/objects/structures/morgue.dm:379]:
```suggestion
			user.visible_message("<span class='notice'>[user] replaces [src]'s igniter.</span>", "<span class='notice'>You replace [src]'s damaged igniter. Now it just needs its paneling welded.</span>")
```

**Comment by procdrone** [comment]
yeah what about the button? destroying it means you can never use the crematorium again.

**Comment by Coolrune206** [comment]
> yeah what about the button? destroying it means you can never use the crematorium again.

As stated in the PR the button and tray are indestructible until a better solution can be found.

**Comment by Coolrune206** [comment]
image
Now comes with a shitty codersprite!

File: code/game/objects/structures/morgue.dm [hunk 2]
```diff
@@ -354,12 +332,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
...
+			return TRUE
+		WELDER_REPAIR_SUCCESS_MESSAGE
+		icon_state = "crema" // there's probably a better way to handle this than setting icon_state, name, and desc to their original values, but I am not a smart man
+		resistance_flags = NONE
+		name = "crematorium"
+		desc = "A human incinerator. Works well on barbeque nights."
+		repairstate = 2
+		broken = FALSE
+		obj_integrity = max_integrity
+		GLOB.crematoriums += src
```
Comment (S34NW) [inline code/game/objects/structures/morgue.dm:368]: You should use `var = initial(var)` here instead of manually defining them

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/morgue.dm (hunk 2 above)
Comment (Coolrune206) [reply code/game/objects/structures/morgue.dm]: done for icon_state, name, desc

File: code/game/objects/structures/morgue.dm [hunk 3]
```diff
@@ -354,12 +332,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
 	add_fingerprint(user)
 	update_icon(UPDATE_OVERLAYS)
 
-/obj/structure/crematorium/attackby(P as obj, mob/user as mob, params)
+/obj/structure/crematorium/obj_break(damage_flag)
+	if(!broken)
```
Comment (Sirryan2002) [inline code/game/objects/structures/morgue.dm:336]: could you make this an early return please

File: code/game/objects/structures/morgue.dm [hunk 4]
```diff
@@ -354,12 +332,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
...
+		update_icon(UPDATE_OVERLAYS)
+		GLOB.crematoriums -= src
+
+/obj/structure/crematorium/welder_act(mob/user, obj/item/I)
+	if(user.a_intent == INTENT_HARM) // if you want to damage it with a welder you should be able to
+		return
+	if(broken)
+		if(!I.tool_use_check(user, 0))
+			return TRUE
+		if(!repairstate == 1)
```
Comment (Sirryan2002) [inline code/game/objects/structures/morgue.dm:354]: could you make the repairstate magic number use defines please?

File: code/game/objects/structures/morgue.dm [hunk 5]
```diff
@@ -354,12 +332,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
...
+		name = initial(name)
+		desc = initial(desc)
+		repairstate = 2
+		broken = FALSE
+		obj_integrity = max_integrity
+		GLOB.crematoriums += src
+		return TRUE
+	else
+		to_chat(user, "<span class='notice'>The crematorium does not seem to need fixing.</span>")
+		return TRUE
```
Comment (Sirryan2002) [inline code/game/objects/structures/morgue.dm:372]: could you move this up to become a `if(!broken)` check and early return so the code above can be unindented

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/morgue.dm (hunk 5 above)
Comment (Coolrune206) [reply code/game/objects/structures/morgue.dm]: donezo

File: code/game/objects/structures/morgue.dm (hunk 3 above)
Comment (Coolrune206) [reply code/game/objects/structures/morgue.dm:340]: Might I ask what you mean here?

File: code/game/objects/structures/morgue.dm [hunk 6]
```diff
@@ -354,12 +336,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
 	add_fingerprint(user)
 	update_icon(UPDATE_OVERLAYS)
 
-/obj/structure/crematorium/attackby(P as obj, mob/user as mob, params)
+/obj/structure/crematorium/obj_break(damage_flag)
```
Comment (S34NW) [inline code/game/objects/structures/morgue.dm:339]: Can you still open and close it when broken? If not, you should probably spit out the contents to stop it being used as some weird unescapeable cell

Comment (Coolrune206) [reply code/game/objects/structures/morgue.dm:339]: It opens and closes just fine while broken, it just can't be turned on.

**Review by S34NW — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]
code looks good

File: code/game/objects/structures/morgue.dm [hunk 7]
```diff
@@ -354,12 +336,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
...
+
+/obj/structure/crematorium/welder_act(mob/user, obj/item/I)
+	if(user.a_intent == INTENT_HARM) // if you want to damage it with a welder you should be able to
+		return
+	if(!broken)
+		to_chat(user, "<span class='notice'>The crematorium does not seem to need fixing.</span>")
+		return TRUE
+	if(!I.tool_use_check(user, 0))
+		return TRUE
+	if(!repairstate == CREMATOR_IN_REPAIR)
```
Comment (SteelSlayer) [inline code/game/objects/structures/morgue.dm:361]: This logic doesn't work properly.
```suggestion
	if(repairstate != CREMATOR_IN_REPAIR)
```

**Review by SteelSlayer — CHANGES_REQUESTED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Comment by Sirryan2002** [comment]
I just noticed some issues with the broken sprite, it appears that it only has the NORTH sprite as broken

Also the shadows in those tears could really use some shading and work

**Pull request merged**